client/extensions/woocommerce/state/sites/shipping-zone-locations/actions.js

```diff
@@ -20,7 +20,7 @@ export const fetchShippingZoneLocations = ( siteId, zoneId ) => ( dispatch, getS
 		return;
 	}
 
-	//0 is Rest of the World zone, skip fetching locations
+	//0 is Locations not covered by your other zones zone, skip fetching locations
```

Comment:
Sorry for the delay, I was neck-deep fixing the saving of Shipping Methods. I've fixed it in https://github.com/Automattic/wp-calypso/pull/15921/commits/22ee744f1d83fb34d84b1a40ed3f999b1b30ac71. I'm OK leaving the comments with the `Locations not covered by your other zones`, I guess I'll get used to it :)

Additionally, I've created https://github.com/woocommerce/woocommerce/pull/16066 to fix this issue in WC Core.